src/proxyvisor.coffee

```diff
@@ -187,24 +203,44 @@ exports.fetchAndSetTargetsForDependentApps = (state, fetchFn) ->
 			Promise.all _.map state.devices, (device, uuid) ->
 				# Only consider one app per dependent device for now
 				appId = _(device.apps).keys().first()
-				knex('dependentDevice').update({ targetEnv: JSON.stringify(device.environment), targetCommit: state.apps[appId].commit }).where({ uuid })
+				knex('dependentDevice').update({
+					targetEnvironment: JSON.stringify(device.environment),
+					targetConfig: JSON.stringify(device.config),
+					targetCommit: state.apps[appId].commit
+				}).where({ uuid })
 	.catch (err) ->
 		console.error('Error fetching dependent apps', err, err.stack)
 
-sendUpdate = (device) ->
-	request.putAsync "#{config.proxyvisorHookReceiver}/v1/devices/#{device.uuid}", {
+sendUpdate = (device, endpoint) ->
+	request.putAsync "#{endpoint}#{device.uuid}", {
 		json: true
 		body:
+			appId: device.appId
 			commit: device.targetCommit
-			environment: JSON.parse(device.targetEnv)
+			environment: JSON.parse(device.targetEnvironment)
+			config: JSON.parse(device.targetConfig)
 	}
 	.spread (response, body) ->
 		if response.statusCode != 200
 			return console.log("Error updating device #{device.uuid}: #{response.statusCode} #{body}")
-		knex('dependentDevice').update({ env: device.targetEnv, commit: device.targetCommit }).where({ uuid: device.uuid })
+		knex('dependentDevice').update({ environment: device.targetEnvironment, commit: device.targetCommit, config: device.targetConfig }).where({ uuid: device.uuid })
 
 exports.sendUpdates = ->
-	# Go through knex('dependentDevice') and sendUpdate if targetCommit or targetEnv differ from the current ones.
-	knex('dependentDevice').select()
-	.map (device) ->
-		sendUpdate(device) if device.targetCommit != device.commit or not _.isEqual(device.targetEnv, device.env)
+	knex('dependentApp').select()
+	.map (app) ->
+		dockerUtils.getImageEnv(app.imageId)
+		.then (imageEnv) ->
+			return imageEnv.RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS if imageEnv.RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS?
```

Comment:
I'm not sure if it's nicer but this could be
```coffee
return imageEnv.RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS ? app.config.RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS ? "#{config.proxyvisorHookReceiver}/v1/devices/"
```
or
```coffee
return imageEnv.RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS ?
	app.config.RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS ?
	"#{config.proxyvisorHookReceiver}/v1/devices/"
```